src/pages/admin/absences/[id].tsx

```diff
@@ -1,45 +1,96 @@
 import ShiftAbsenceShow from "@/components/ShiftAbsenceShow";
 import StatusButton from "@/components/StatusButton";
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import { GetAbsenceForAdmin200ResponseData as Absence } from "@/api/typescript-axios";
 
-// 仮の時間を設定
-const startTime = new Date("5 17, 1998 18:00:00");
-const endTime = new Date("5 17, 1998 23:00:00");
+export default function AdminAbsencesShow() {
+  const router = useRouter();
+  const [isLoading, setLoading] = useState<boolean>(true);
+  const [absence, setAbsence] = useState<Absence>();
+  const [updateAbsenceStatus, setUpdateAbsenceStatus] = useState<boolean>();
+  // データフェッチの際、idはNumber型で指定している
+  const id = Number(router.query.id);
 
-const rows = [
-  {
-    detailLabel: "従業員名",
-    detail: "I am 従業員",
-  },
-  {
-    detailLabel: "開始時間",
-    detail: startTime.toLocaleString(),
-  },
-  {
-    detailLabel: "終了時間",
-    detail: endTime.toLocaleString(),
-  },
-  {
-    detailLabel: "承認状況",
-    detail: "未承認",
-  },
-];
+  useEffect(() => {
+    (async () => {
+      // router.isReady はboolean。trueでないと、idがundefinedになってしまうため、trueの時にデータフェッチを行う。
+      if (router.isReady) {
+        try {
+          const absenceRes = await KintaiAdmin.getAbsenceForAdmin(id);
+          setAbsence(absenceRes.data.data);
+        } catch (error) {
+          // TODO: エラー処理と、その共通化
+        }
+        setLoading(false);
+      }
+    })();
+  }, [router.isReady, updateAbsenceStatus]);
+
+  const statusJa = {
+    unapproved: "未承認",
+    approved: "承認",
+    rejected: "却下",
+  };
+
+  const rows = absence
+    ? [
+        {
+          detailLabel: "従業員名",
+          detail: absence.employee.name,
+        },
+        {
+          detailLabel: "開始時間",
+          detail: absence.startTime.toLocaleString(),
+        },
+        {
+          detailLabel: "終了時間",
+          detail: absence.endTime.toLocaleString(),
+        },
+        {
+          detailLabel: "承認状況",
+          detail: statusJa[`${absence.status}`],
+        },
+      ]
+    : [];
+
+  type StatusProps = "approved" | "rejected";
+
+  const changeStatus = async (afterStatus: StatusProps) => {
+    await KintaiAdmin.updateAbsenceStatus(id, {
+      status: afterStatus,
+    });
+    setUpdateAbsenceStatus(!updateAbsenceStatus);
+  };
 
-export default function AdminAbsencesShow() {
   return (
     <>
       <MyHead title="欠勤詳細" />
       <AdminHeader />
       <main className="text-center">
         <h2 className="mt-5 mb-5">欠勤詳細</h2>
-        {/* TODO : absenceが存在しなければ
-        <p>該当する欠勤は存在しません</p> */}
-        <ShiftAbsenceShow rows={rows} />
-        <div className="d-flex justify-content-center col-8 mx-auto text-center">
-          <StatusButton link="/" statusLabel="承認" />
-          <StatusButton link="/" statusLabel="却下" />
-        </div>
+        {isLoading ? (
+          <>Loading...</>
+        ) : absence ? (
+          <>
+            <ShiftAbsenceShow rows={rows} />
```

Comment:
`[id].tsx` が結構複雑になってきてるので、 absence をそのまま渡せるようにしてみてはいかがでしょうか

```suggestion
            <ShiftAbsenceShow absence={absence} />
```

Reply:
@yuto-ono 
https://github.com/tatata05/next-app/pull/22/commits/3fea62de027e6dc1414ac341219b91cf151f37b1
absence をそのまま渡す形に変更しました！